### PR TITLE
Approve AToken once

### PR DESCRIPTION
**Pull request by aodhgan**
as per https://github.com/code-423n4/2021-06-pooltogether-findings/issues/93

### CODE REVIEW

File: test/ATokenYieldSource.test.ts [hunk 1]
```diff
@@ -97,7 +98,9 @@ describe('ATokenYieldSource', () => {
         hardhatATokenYieldSourceHarness.address,
         contractsOwner,
       )) as ATokenYieldSourceHarness;
-  
+    
+    await underlyingToken.mock.safeApprove.returns(true);
```
Comment (PierrickGT) [inline test/ATokenYieldSource.test.ts]: To mock `safeApprove`, you actually need to mock `approve`, so something like:
```suggestion
    await underlyingToken.mock.approve.withArgs(lendingPoolAddress, maxAmount).returns(true);
```

**Review by PierrickGT — CHANGES_REQUESTED** [review]

**Review by PierrickGT — APPROVED** [review]

**Review by asselstine — CHANGES_REQUESTED** [review]
We also need a function to re-approve the max amount, in case for whatever reason the approval level drops too low.

**Comment by PierrickGT** [comment]
> We also need a function to re-approve the max amount, in case for whatever reason the approval level drops too low.

@asselstine added in this commit: https://github.com/pooltogether/aave-yield-source/pull/21/commits/efe08df871f5437f5a9899a6617b188fc7d355f7

**Review by asselstine — APPROVED** [review]
...with a test!  Good job.

**Pull request merged**